fuel-core-interfaces/src/relayer.rs

```diff
@@ -7,16 +7,25 @@ use tokio::sync::oneshot;
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 #[derive(Debug, Clone)]
 pub struct StakingDiff {
-    /// Validator registration, it can be new consensus address if registered or
-    /// None if unregistration happened.
-    pub validators: HashMap<Address, Option<Address>>,
+    /// Validator registration, it is pair of old consensu key and new one, where consensus address
+    /// if registered is Some or None if unregistration happened.
+    pub validators: HashMap<Address, ValidatorDiff>,
     /// Register changes for all delegations inside one da block.
     pub delegations: HashMap<Address, Option<HashMap<Address, ValidatorStake>>>,
 }
 
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone)]
+pub struct ValidatorDiff {
+    /// Previous consensus key, None if validator was not set.
+    pub previous_consensus_key: Option<Address>,
```

Comment:
[nit] would it be more correct to call this a consensus_address, or should this type really be a Option<PublicKey>?

```suggestion
    pub previous_consensus_key: Option<PublicKey>,
```

Reply:
Keep in mind we use uncompressed public keys, so they need 64 bytes while address is only 32

Reply:
New type will ba made